modules/gapi/test/infer/gapi_infer_ie_test.cpp

```diff
@@ -937,6 +953,339 @@ TEST_F(ROIListNV12, Infer2MediaInputNV12)
+    std::vector<cv::Mat> gapi_genders;
+
+    std::vector<cv::Rect> roi_list = {
+        cv::Rect(cv::Point{64, 60}, cv::Size{ 96,  96}),
+        cv::Rect(cv::Point{50, 32}, cv::Size{128, 160}),
+    };
+
+    using AGInfo = std::tuple<cv::GMat, cv::GMat>;
+    G_API_NET(AgeGender, <AGInfo(cv::GMat)>, "test-age-gender");
+
+    cv::GArray<cv::Rect> rr;
+    cv::GMat in;
+    cv::GArray<cv::GMat> age, gender;
+    std::tie(age, gender) = cv::gapi::infer2<AgeGender>(in, rr);
+
+    cv::GComputation comp(cv::GIn(in, rr), cv::GOut(age, gender));
+
+    auto pp = cv::gapi::ie::Params<AgeGender> {
+        params.model_path, params.weights_path, params.device_id
+    }.cfgOutputLayers({ "age_conv3", "prob" });
+
+
+    std::size_t num_frames = 0u;
+    std::size_t max_frames = 10u;
+
+    cv::VideoCapture cap;
+    cap.open(filepath);
+    if (!cap.isOpened())
+        throw SkipTestException("Video file can not be opened");
+
+    cap >> in_mat;
+    auto pipeline = comp.compileStreaming(cv::compile_args(cv::gapi::networks(pp)));
+    pipeline.setSource(
+            cv::gin(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(filepath), roi_list));
+
+    pipeline.start();
+    while (num_frames < max_frames && pipeline.pull(cv::gout(gapi_ages, gapi_genders)))
+    {
+        {
+            auto plugin        = cv::gimpl::ie::wrap::getPlugin(params);
+            auto net           = cv::gimpl::ie::wrap::readNetwork(params);
+            setNetParameters(net);
+            auto this_network  = cv::gimpl::ie::wrap::loadNetwork(plugin, net, params);
+            auto infer_request = this_network.CreateInferRequest();
+            auto frame_blob = cv::gapi::ie::util::to_ie(in_mat);
+
+            for (auto &&rc : roi_list) {
+                const auto ie_rc = IE::ROI {
+                    0u
+                    , static_cast<std::size_t>(rc.x)
+                    , static_cast<std::size_t>(rc.y)
+                    , static_cast<std::size_t>(rc.width)
+                    , static_cast<std::size_t>(rc.height)
+                };
+                infer_request.SetBlob("data", IE::make_shared_blob(frame_blob, ie_rc));
+                infer_request.Infer();
+
+                using namespace cv::gapi::ie::util;
+                ie_ages.push_back(to_ocv(infer_request.GetBlob("age_conv3")).clone());
+                ie_genders.push_back(to_ocv(infer_request.GetBlob("prob")).clone());
+            }
+        } // namespace IE = ..
+        // Validate with IE itself (avoid DNN module dependency here)
+        normAssert(ie_ages   [0], gapi_ages   [0], "0: Test age output");
+        normAssert(ie_genders[0], gapi_genders[0], "0: Test gender output");
+        normAssert(ie_ages   [1], gapi_ages   [1], "1: Test age output");
+        normAssert(ie_genders[1], gapi_genders[1], "1: Test gender output");
+
+        ie_ages.clear();
+        ie_genders.clear();
+
+        ++num_frames;
+        cap >> in_mat;
+    }
+    pipeline.stop();
+}
+
```

Comment:
I would recommend to add some kind of stress test to ensure that there are no race conditions in the code.

Reply:
Good idea, but what kind of test ? I implement tests which process 10 frames it checks race condition, doesn't it ?